src/Microsoft.ApplicationInsights.AspNetCore/RequestTrackingTelemetryModule.cs

```diff
@@ -103,10 +101,14 @@ void IObserver<DiagnosticListener>.OnNext(DiagnosticListener value)
 
             foreach (var applicationInsightDiagnosticListener in this.diagnosticListeners)
             {
-                if (applicationInsightDiagnosticListener.ListenerName == value.Name)
+                if (applicationInsightDiagnosticListener is HostingDiagnosticListener)
                 {
-                    subs.Add(value.Subscribe(applicationInsightDiagnosticListener));
-                    applicationInsightDiagnosticListener.OnSubscribe();
+                    if (applicationInsightDiagnosticListener.ListenerName == value.Name)
+                    {
+                        Predicate<string> hostingPredicate = (string eventName) => (eventName != null) ? !(eventName[21] == 'M') || eventName == "Microsoft.AspNetCore.Mvc.BeforeAction" : false;
```

Comment:
I guess it worth a comment :)

Reply:
nit: hosting predicate can be made static field - it seems there is no need to create it here on demand - we know we'll need it